### ISSUE TITLE
Mancano le porte tra le stanze

### ISSUE THREAD

**Issue by frigierim** [frigierim/ggj-tracking#14]
Creare un componente da inserire tra le stanze che blocchi l'accesso finchè non viene attivato.

**Comment by CristiJ92** [comment]
v.0.0.8:
le porte le hai piazzate in Godot o sempre in Tiles?

**Comment by frigierim** [comment]
Tutta la roba "interattiva" adesso viene messa giù in Godot (note, porte, passaggi tra i livelli). In Tiled si disegna solamente il livello e gli ostacoli fissi (muri, armadi, tavoli, sedie).

**Comment by CristiJ92** [comment]
Le porte devono essere attivate con il tasto conferma del giocatore attuale

**Comment by frigierim** [comment]
Dalla versione 0.0.20, quando il giocatore si trova in prossimità di una nota o di una porta, questa lampeggerà e il giocatore potrà interagire con essa usando il proprio tasto di conferma.

**Comment by CristiJ92** [comment]
puoi disattivare la collisione con la porta quando è aperta?

**Comment by frigierim** [comment]
Intendi in modo che non si possa più richiudere? Mi sembrava utile per creare uno svantaggio all'altro giocatore.

**Comment by CristiJ92** [comment]
no, per evitare che il giocatore venga spostato dal movimento della porta...

**Comment by frigierim** [comment]
Penso si possa fare, ma così il giocatore passerà "attraverso" alla porta. Se è ok, procedo.

**Comment by CristiJ92** [comment]
la collisione si deve disattivare a porta aperta e si riattiva alla chiusura, prova a gestire questa cosa... poi vediamo se tenere modificare o fare reverse

**Comment by frigierim** [comment]
Ok, quello si può fare... ciò che intendevo io è che DURANTE l'apertura della porta il giocatore ci passerà attraverso. Comunque, provo a metterlo in piedi e poi vediamo se ci piace o no.

**Comment by CristiJ92** [comment]
meh...

v.0.0.25

a livello di design non mi convince.. tanto meno le possibili soluzioni che mi vengono in mente... 
direi di provare a fare il reverse della collisione avendo però la porta che rileva da che lato ha avuto l'interazione e quindi ruotare nel senso opposto

es. a partire da:
![image](https://user-images.githubusercontent.com/45758943/55396284-d074dc00-5543-11e9-9918-bc1deb572d0c.png)

porta chiusa->azione->porta si apre verso l'altro (_0°_**+90°**)->avanzare verso N->azione->porta si chiude (_+90_**°-90°**)->azione->porta si apre verso il basso(_0°_**-90°**)->avanzare verso S->azione->porta si chiude(_-90°_**+90°**)

**Comment by frigierim** [comment]
Ok, provo a implementare le porte da saloon :)

**Comment by frigierim** [comment]
Fatto, ma anche così mi sembra un po' meh. Ho aggiunto una flag nella schermata dei cheat per abilitare/disabilitare il fatto che le porte entrino in collisione col giocatore mentre si aprono, per poter vedere in presa diretta entrambe le versioni e decidere se ce n'è una che fa meno pietà.

**Comment by CristiJ92** [comment]
V.0.0.26:
da porta aperta: la collisione si dovrebbe attivare prima della rotazione... così dovrebbe essere decente.

EDIT: hai già fatto... uncheck ghost doors, si dovrà aggiustare in futuro l'interazione con le porte

reminder: aggiungere al GDD

**Comment by frigierim** [comment]
Ok, allora di default metto il check disabilitato.

**Comment by CristiJ92** [comment]
> V.0.0.26:
> da porta aperta: la collisione si dovrebbe attivare prima della rotazione... così dovrebbe essere decente.
> 
> EDIT: hai già fatto... uncheck ghost doors, **si dovrà aggiustare in futuro l'interazione con le porte**
> 
> reminder: aggiungere al GDD

V.0.1.1
verificato, ok. 
Aspettare l'aggiornamento del GDD per fixare l'interazione
aggiungo tag TODO

**Comment by CristiJ92** [comment]
aggiornato GDD
![image](https://user-images.githubusercontent.com/45758943/56866606-70a31100-69db-11e9-923c-a8270e1fde25.png)
adesso abbiamo bisogno di un po di assets per le porte/porte di ingresso/finestre bloccati e sbloccati, creo issue apposito
